src/main/java/org/intellij/sonar/sonarserver/SonarServer.java

```diff
@@ -87,109 +98,65 @@ private SonarClient createSonarClient(Host host) {
-        wsUtils.getString(json, "lang"), wsUtils.getString(json, "langName"), wsUtils.getString(json, "htmlDesc"),
-        wsRule.description());
-        return rule;
+        Rules.Rule rule = sonarClient.rules().show(null, key).getRule();
+        return new Rule(rule.getKey(), rule.getName(), rule.getSeverity(), rule.getLang(), rule.getLangName(), rule.getHtmlDesc());
     }
 
     public List<Resource> getAllProjectsAndModules() {
-        List<Resource> allResources = new LinkedList<Resource>();
+        List<Resource> allResources = new LinkedList<>();
         final ProgressIndicator indicator = ProgressManager.getInstance().getProgressIndicator();
         indicator.setText("Downloading SonarQube projects");
-        List<Resource> projects = getAllProjects(sonar);
-        projects = projects.stream().sorted(new ByResourceName()).collect(Collectors.toList());
+        List<Component> projects = getAllProjects(sonarClient);
+        projects = projects.stream().sorted(comparing(Component::getName)).collect(toList());
 
         if (null != projects) {
             indicator.setText("Downloading SonarQube modules");
             int i = 0;
-            for (Resource project : projects) {
+            for (Component project : projects) {
                 if (indicator.isCanceled()) break;
                 i++;
                 indicator.setFraction(1.0 * i / projects.size());
                 indicator.setText2(project.getName());
-                allResources.add(project);
-                List<Resource> modules = getAllModules(sonar, project.getId());
-                modules = modules.stream().sorted(new ByResourceName()).collect(Collectors.toList());
+                allResources.add(new Resource(project.getKey(), project.getName(), project.getName(), "PRJ", project.getQualifier(), project.getDescription()));
+                List<Component> modules = getAllModules(sonarClient, project.getId());
+                modules = modules.stream().sorted(comparing(Component::getName)).collect(toList());
                 if (null != modules) {
-                    for (Resource module : modules) {
-                        allResources.add(module);
+                    for (Component module : modules) {
+                        allResources.add(new Resource(module.getKey(), module.getName(), module.getName(), "PRJ", module.getQualifier(), module.getDescription()));
                     }
                 }
             }
         }
         return allResources;
     }
 
-    public List<Resource> getAllProjects(Sonar sonar) {
-        ResourceQuery query = new ResourceQuery();
-        query.setQualifiers(Resource.QUALIFIER_PROJECT);
-        query.setTimeoutMilliseconds(READ_TIMEOUT_IN_MILLISECONDS);
-        return sonar.findAll(query);
+    public List<Component> getAllProjects(WsClient sonarClient) {
+        org.sonarqube.ws.client.component.SearchWsRequest query = new org.sonarqube.ws.client.component.SearchWsRequest()
```

Comment:
can the full names be replaced by imports?

Reply:
Nope, multiple likewise-named classes used in there, only one can be imported, unfortunately this is not Groovy. :-)